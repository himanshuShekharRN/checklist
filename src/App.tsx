import React from 'react';
import 'react-native-gesture-handler';

import {Navigation} from './navigation';
import {Provider} from 'react-redux';
import {persistor, store} from './store';
import {PersistGate} from 'redux-persist/integration/react';

export const App: React.FC = () => {
  return (
    <Provider store={store}>
      <PersistGate loading={null} persistor={persistor}>
        <Navigation />
      </PersistGate>
    </Provider>
  );
};
