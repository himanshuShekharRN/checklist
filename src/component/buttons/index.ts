import {FloatingAddButton} from './floatingAddButton';
import {RoundedButton} from './roundedButton';

export {FloatingAddButton, RoundedButton};
