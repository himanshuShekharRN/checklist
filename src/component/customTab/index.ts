import {CustomTab} from './CustomTab';
import {FirstRoute} from './firstRoute';
import {SecondRoute} from './secondRoute';
import {ThirdRoute} from './thirdRoute';
import {FourthRoute} from './fourthRoute';
import {FifthRoute} from './fifthRoute';

export {
  CustomTab,
  FirstRoute,
  SecondRoute,
  ThirdRoute,
  FourthRoute,
  FifthRoute,
};
