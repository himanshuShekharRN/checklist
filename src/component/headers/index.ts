import {HeadersWithButton} from './headersWithButton/HeadersWithButton';

export {HeadersWithButton};
