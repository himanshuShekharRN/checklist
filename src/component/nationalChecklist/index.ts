import {NationalChecklist} from './NationalChecklist';

export {NationalChecklist};
