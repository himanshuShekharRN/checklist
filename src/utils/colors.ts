export const COLOR_BLACK = '#000000';
export const COLOR_WHITE = '#FFFFFF';
export const COLOR_NAVY_300 = '#ECEEF0';
