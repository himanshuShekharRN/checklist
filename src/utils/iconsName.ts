export const OUTLINE_EXCLAMATION = 'error-outline';
export const WORKSPACE_PREMIUM = 'workspace-premium';
export const FILE_FILLED = 'description';
export const DONE = 'check-circle';
export const SKIPPED_OPTIONAL = 'do-disturb-on';
export const PENDING = 'info';
export const UNCHECK = 'panorama-fish-eye';
export const DELETE = 'delete';
export const RIGHT_ANGLE_BRACKET = 'keyboard-arrow-right';
