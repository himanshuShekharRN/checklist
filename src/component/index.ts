import {SwipableCards} from './swipableCards/SwipableCards';
import {SwipableActionButton} from './swipableActionButton/SwipableActionButton';
import {Card} from './cards';
import {MyCheckList} from './myChecklist';
import {Space} from './space';
import {PreDepartureList} from './preDepartureList';
import {Title} from './title';
import {HeadersWithButton} from './headers';
import {CustomSVG} from './customSVG';
import {CustomIcon} from './customIcon';
import {FloatingAddButton, RoundedButton} from './buttons';
import {ListCard} from './listCard';
import {LinearProgressBar} from './progressBar';
import {CustomTab} from './customTab';
import {DepartureDocCard} from './departureDocCard';
import {NationalChecklist} from './nationalChecklist';

export {
  FloatingAddButton,
  Card,
  MyCheckList,
  Space,
  PreDepartureList,
  Title,
  HeadersWithButton,
  CustomSVG,
  CustomIcon,
  RoundedButton,
  ListCard,
  LinearProgressBar,
  CustomTab,
  DepartureDocCard,
  NationalChecklist,
  SwipableActionButton,
  SwipableCards,
};
