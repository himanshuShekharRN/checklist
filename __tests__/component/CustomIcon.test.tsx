import React from 'react';
import {render, fireEvent, screen} from '@testing-library/react-native';
import {CustomIcon} from '../../src/component';
import {RIGHT_ANGLE_BRACKET} from '../../src/utils/iconsName';
import {COLOR_BLACK} from '../../src/utils/colors';

describe('CustomIcon', () => {
  const onPressHandlerMock = jest.fn();
  const iconWrapperStyle = {
    flex: 1,
  };
  const props = {
    size: 16,
    name: RIGHT_ANGLE_BRACKET,
    color: COLOR_BLACK,
    onPressHandler: onPressHandlerMock,
    isPressable: true,
    iconWrapperStyle: iconWrapperStyle,
  };

  const tree: any = render(<CustomIcon {...props} />);

  test('renders correctly', () => {
    expect(tree.toJSON()).toBeTruthy();
  });

  test('calls onPressHandler when the customIcon is pressable and pressed', () => {
    render(<CustomIcon {...props} />);

    const customIcon = screen.getByTestId('customIcon');
    fireEvent.press(customIcon);

    expect(onPressHandlerMock).toHaveBeenCalled();
  });

  test('applies custom iconWrapperStyle', () => {
    render(<CustomIcon {...props} />);

    const customIcon = screen.getByTestId('customIcon');
    expect(customIcon.props.style).toBe(iconWrapperStyle);
  });

  test('snapshot created successfully', () => {
    render(<CustomIcon {...props} />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
