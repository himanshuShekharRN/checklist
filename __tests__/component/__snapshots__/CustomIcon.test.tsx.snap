// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomIcon snapshot created successfully 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "flex": 1,
    }
  }
  testID="customIcon"
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      [
        {
          "color": "#000000",
          "fontSize": 16,
        },
        undefined,
        {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        {},
      ]
    }
  >
    îŒ•
  </Text>
</View>
`;
